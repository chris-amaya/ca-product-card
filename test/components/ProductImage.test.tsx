import * as React from 'react';

import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductImage', () => {
  it('debe de mostrar el componente correctamente', () => {
    const wrapper = renderer.create(<ProductImage img="https://hola.jpg" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('debe de mostrar el componente con imagen personalizada', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
