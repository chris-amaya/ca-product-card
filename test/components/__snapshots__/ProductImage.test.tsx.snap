// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente con imagen personalizada 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="coffee mug"
    className="productImg undefined"
    src="https://placehold.it/200x150"
  />
</div>
`;

exports[`ProductImage debe de mostrar el componente correctamente 1`] = `
<img
  alt="coffee mug"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;
